include/openmc/particle_data.h

```diff
@@ -0,0 +1,453 @@
+#ifndef OPENMC_PARTICLE_REPRESENTATION_H
+#define OPENMC_PARTICLE_REPRESENTATION_H
+
+#include "openmc/array.h"
+#include "openmc/constants.h"
+#include "openmc/position.h"
+#include "openmc/random_lcg.h"
+#include "openmc/tallies/filter_match.h"
+#include "openmc/vector.h"
+
+#ifdef DAGMC
+#include "DagMC.hpp"
+#endif
+
+namespace openmc {
+
+//==============================================================================
+// Constants
+//==============================================================================
+
+// Since cross section libraries come with different numbers of delayed groups
+// (e.g. ENDF/B-VII.1 has 6 and JEFF 3.1.1 has 8 delayed groups) and we don't
+// yet know what cross section library is being used when the tallies.xml file
+// is read in, we want to have an upper bound on the size of the array we
+// use to store the bins for delayed group tallies.
+constexpr int MAX_DELAYED_GROUPS {8};
+
+constexpr double CACHE_INVALID {-1.0};
+
+//==========================================================================
+// Aliases and type definitions
+
+//! Particle types
+enum class ParticleType { neutron, photon, electron, positron };
+
+//! Saved ("banked") state of a particle
+//! NOTE: This structure's MPI type is built in initialize_mpi() of
+//! initialize.cpp. Any changes made to the struct here must also be
+//! made when building the Bank MPI type in initialize_mpi().
+//! NOTE: This structure is also used on the python side, and is defined
+//! in lib/core.py. Changes made to the type here must also be made to the
+//! python defintion.
+struct ParticleBank {
```

Comment:
I've actually been wanting to separate out this type and rename it as well. I was thinking `SourceSite` might be a better indication of its use in most places. What do you think? It is the type that `Source::sample` returns after all.

Reply:
Yes, I think that's a far more descriptive name. I'll definitely add that in. I like the guidance that if you use a struct outside of a class, it should probably not be embedded in that class.